### DISCUSSION TITLE
How can I get the Twitter oauth token secret?

### DISCUSSION THREAD

**Discussion by gabrielperales** [supabase/supabase#5317] [Questions] [answered]
I'm trying to make an app that uses the Twitter API. To twit on behalf of somebody I need the `oauth_token` and the `oauth_token_secret`.  When I signup with Twitter using Supabase Auth, I can get the `oauth_token` from the session like [here](https://supabase.com/docs/reference/javascript/auth-signin#sign-in-with-scopes) is described. 

But, how can I get the `oauth_token_secret`? 

I've been playing with next-auth https://github.com/nextauthjs/adapters/pull/188 to make a workaround, but then I lose the Supabase Auth and I have to make a lot of extra work to make the API REST again because I can't use the RLS.

**Comment by maximedupre** [comment]
Did you end up finding how to access the secret token? 🤔

**Comment by maciekChmura** [comment]
Hello @gabrielperales I'm also experimenting and would like to build a Twitter app like you described. 
Have you managed to get the `oauth_token_secret`?

**Comment by awalias** [comment, marked as answer]
supabase unfortunately can't return the oauth token secret for twitter during this authentication step because twitter uses [OAuth1a for this API](https://developer.twitter.com/en/docs/authentication/oauth-1-0a)

**Comment by diogoribeirodev** [comment]
So i guess use NextAuth is the only solution.
